includes/DeactivationSurvey.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace NewfoldLabs\WP\Module\Data;
+
+use function NewfoldLabs\WP\ModuleLoader\container;
+
+/**
+ * Class DeactivationSurvey
+ *
+ * Brand plugins deactivation survey modal.
+ *
+ * @package NewfoldLabs\WP\Module\Data
+ */
+class DeactivationSurvey {
+
+    /**
+     * DeactivationSurvey constructor.
+     */
+    public function __construct() {
+        $this->deactivation_survey_assets();
+        $this->deactivation_survey_runtime();
+    }
+
+    /**
+     * Enqueue deactivation survey assets.
+     */
+    public function deactivation_survey_assets() {
+        $assetsDir = container()->plugin()->url . 'vendor/newfold-labs/wp-module-data/includes/assets/';
+
+        // accessible a11y dialog
+        wp_register_script(
+            'nfd-data-a11y-dialog',
+            $assetsDir . 'js/a11y-dialog.min.js',
+            array(),
+            '8.0.4',
+        );
+
+        // deactivation-survey.js
+        wp_enqueue_script(
+            'nfd-data-deactivation-survey',
+            $assetsDir . 'js/deactivation-survey.js',
+            array( 'nfd-data-a11y-dialog' ),
+            container()->plugin()->version,
+            true
+        );
+
+        // Styles
+		wp_enqueue_style(
+			'nfd-data-deactivation-survey-style',
+			$assetsDir . 'css/deactivation-survey.css',
+			array(),
+			container()->plugin()->version
+		);
+    }
+
+    /**
+     * Localize deactivation survey runtime.
+     */
+    public function deactivation_survey_runtime() {
+        $plugin_slug = explode( '/', container()->plugin()->basename )[0];
+
+        wp_localize_script(
+			'nfd-data-deactivation-survey',
+			'newfoldDataDeactivationSurvey',
+			array(
+				'eventsEndpoint' => \esc_url_raw( \get_home_url() . '/index.php?rest_route=/newfold-data/v1/events/' ),
```

Comment:
We should use 
```suggestion
				'eventsEndpoint' => \esc_url_raw( \rest_url() . '/newfold-data/v1/events/' ),
```
this ensures the appropriate permalink structure as appropriate